app/templating/summary/group.py

```diff
@@ -1,29 +1,63 @@
+from collections import defaultdict
+
 from app.templating.summary.block import Block
+from app.helpers.schema_helpers import get_group_instance_id
+from app.questionnaire.location import Location
+from app.templating.template_renderer import renderer
 
 
 class Group:
 
-    def __init__(self, group_schema, path, answer_store, metadata, schema):
-        self.id = group_schema['id']
-        self.title = group_schema.get('title')
-        self.blocks = self._build_blocks(group_schema, path, answer_store, metadata, schema)
+    def __init__(self, group_schema, path, answer_store, metadata, schema, group_instance, schema_context):
+        self.id = group_schema['id'] + '-' + str(group_instance)
+        location = Location(group_schema['id'], group_instance, group_schema['blocks'][0]['id'])
+        self.schema_context = schema_context
+        self.schema_context['group_instance_id'] = get_group_instance_id(schema, answer_store, location)
+
+        self.open_summary_block = False
+        self.close_summary_block = False
```

Comment:
Are these 2 more display logic? rather than the model?

Reply:
We could use`{{loop.first}}` and `{{loop.last}}` in the templates?
http://jinja.pocoo.org/docs/2.10/templates/

Reply:
That's really nice!

Reply:
I can't see an obvious way to replicate the logic in build_summary_rendering_context without changing the context drastically. Thoughts?

Reply:
Because we only want to close/open when the title changes?
`{{loop.changed(group.title)}}`